src/foam/u2/filter/StringFilterView.js

```diff
@@ -113,30 +118,19 @@ foam.CLASS({
       required: true
     },
     {
-      name: 'daoContents',
-      preSet: function(_, n) {
-        // remove objects with the same strings for specified property
-        var self = this;
-        return n.reduce(function(accumulator, obj) {
-
-          // create an identifying id
-          var id = obj[self.property];
-
-          // if the id is not found in the temp array
-          // add the object to the output array
-          // and add the key to the temp array
-          if ( accumulator.temp.indexOf(id) === -1 ) {
-            accumulator.out.push(obj);
-            accumulator.temp.push(id);
-          }
-          return accumulator;
-        // return the zero duplicate array
-        }, { temp: [], out: [] }).out;
-      }
+      name: 'daoContents'
     },
     {
       class: 'String',
-      name: 'search'
+      name: 'search',
+      postSet: function(_, n) {
+        this.dao.where(this.CONTAINS_IC(this.property, n)).limit(100).select(this.GroupBy.create({
```

Comment:
What is the limit(100) for? You know that this will not give you all values in most cases?

Reply:
This is supposed to be a set of possible values to filter by, so it seems reasonable to limit the set of values to filter by to a reasonable number of results that the user is likely to scroll through without causing performance problems on the client .. if they want more specific results, they should enter a search string to isolate on that result. I don't think an exhaustive list is the way to go here, as a set of unique values in a given column would just return the whole DAO and we'd be back in the same position as before.

Reply:
See Ticket Discussion Comments: https://nanopay.atlassian.net/browse/CPF-3792

Reply:
But the limit() is on the records in the DAO, not the resulting keys, so you might just get the same key 100 times.

Reply:
The MDAO can short-circuit GROUP_BY(COUNT)s.